lib/sanbase/auth/settings.ex

```diff
@@ -69,6 +72,21 @@ defmodule Sanbase.Accounts.Settings do
     )
   end
 
+  defp validate_favorite_metrics(_, nil), do: []
+
+  defp validate_favorite_metrics(_, type) do
+    type
+    |> Stream.map(&Sanbase.Metric.has_metric?/1)
+    |> Enum.find(&(&1 != true))
+    |> case do
+      nil ->
+        []
+
+      {:error, error} ->
+        [favorite_metrics: "Invalid metric found in the favorite metrics list. #{error}"]
+    end
```

Comment:
A more simple way would be:
```suggestion
    metrics
    |> Enum.find_value([], fn metric ->
      case Sanbase.Metric.has_metric?(metric) do
        true -> false
        {:error, error} -> [favorite_metrics: "Invalid metric found in the favorite metrics list. #{error}"]
      end
    end)
```